src/api.rs

```diff
@@ -0,0 +1,52 @@
+use std::sync::Arc;
+
+use jsonrpc_core::{Result, Error, ErrorCode};
+use jsonrpc_derive::rpc;
+use substrate_client::{Client, CallExecutor, backend};
+use sr_primitives::{traits::Block as BlockT, traits::ProvideRuntimeApi, generic::BlockId};
+use primitives::Blake2Hasher;
+use centrifuge_chain_runtime::{Hash, AnchorApi, anchor::AnchorData, BlockNumber};
+
+const RUNTIME_ERROR: i64 = 1;
+
+/// Anchor RPC methods.
+#[rpc]
+pub trait AnchorRpcApi {
+    /// Returns the next valid index (aka nonce) for given account.
+    #[rpc(name = "anchor_getAnchorById")]
+    fn get_anchor_by_id(&self, id: Hash) -> Result<AnchorData<Hash, BlockNumber>>;
+}
+
+pub struct Anchors<B, E, Block: BlockT, RA> {
```

Comment:
a unit test for this stuff is on the way.

Reply:
this is pretty complicated to set up the dependancies for the test :| . Since I have func tests update coming for this api, I will add this test later when I clarify how to add a proper test for this.